cdap-app-templates/cdap-etl/cdap-etl-proto/src/main/java/io/cdap/cdap/etl/proto/v2/spec/StageSpec.java

```diff
@@ -315,4 +315,11 @@ public int hashCode() {
       return Objects.hash(port, schema);
     }
   }
+
+  public static StageSpec createCopyWithUpdatedMaxPreviewRecords(StageSpec stageSpec, int newMaxPreviewRecords) {
```

Comment:
Please add javadoc to the public method describing what it does.

Reply:
This method name is unnecessarily long. Just call it `create` or `copy`, which is also easy for future expansion (imagine there are multiple values you want to update the stage spec).